pkg/controller/logstorage/cleanup/cleanup_controller.go

```diff
@@ -0,0 +1,163 @@
+	c, err := controller.New("log-storage-cleanup-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Tenant{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-cleanup-controller failed to watch Tenant resource: %w", err)
+	}
+
+	err = utils.AddPeriodicReconcile(c, utils.PeriodicReconcileTime, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("log-storage-cleanup-controller failed to create periodic reconcile watch: %w", err)
+	}
+
+	return nil
+}
+
+func (r *CleanupController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	helper := utils.NewNamespaceHelper(r.multiTenant, render.ElasticsearchNamespace, request.Namespace)
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
+	reqLogger.Info("Reconciling LogStorage - Cleanup")
+
+	// Wait for Elasticsearch to be installed and available.
+	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred trying to retrieve Elasticsearch", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Elasticsearch cluster to be operational", nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+
+	// Clean up any stale users that may have been left behind by a previous tenant
+	if err := r.cleanupStaleUsers(ctx, reqLogger); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failure occurred while cleaning up stale users", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	r.status.ReadyToMonitor()
+	r.status.ClearDegraded()
+	return reconcile.Result{}, nil
+}
+
+func (r *CleanupController) cleanupStaleUsers(ctx context.Context, logger logr.Logger) error {
+	esClient, err := r.esClientFn(r.client, ctx, relasticsearch.ElasticEndpoint())
+	if err != nil {
+		return fmt.Errorf("failed to connect to Elasticsearch - failed to create the Elasticsearch client")
+	}
+
+	allESUsers, err := esClient.GetUsers(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch users from Elasticsearch")
+	}
+
+	tenants := operatorv1.TenantList{}
+	err = r.client.List(ctx, &tenants)
+	if err != nil {
+		return fmt.Errorf("failed to fetch TenantList")
+	}
+
+	// TODO: Fetch cluster UUID and use it to filter out stale users from clusters other than our own. The exact form
+	// this will take is TBD. However the cluster UUID is stored/fetched should be immutable to prevent the possibility
+	// of a change in cluster-id resulting in orphaned users
+	clusterIDConfigMap := corev1.ConfigMap{
+		ObjectMeta: apiv1.ObjectMeta{
```

Comment:
You don't need to populate the struct prior to calling `Get()` - all of this will be filled in by the `Get` call itself. You can just pass a pointer to an empty struct.